src/xyk.rs

```diff
@@ -56,8 +56,7 @@ pub fn calculate_out_given_in(
     let numerator = out_reserve_hp.checked_mul(amount_in_hp).ok_or(Overflow)?;
     let sale_price_hp = numerator.checked_div(denominator).ok_or(Overflow)?;
 
-    let result = to_balance!(sale_price_hp).ok();
-    round_up!(result.ok_or(Overflow)?)
+    to_balance!(sale_price_hp)
```

Comment:
I think we should add the same check of zero amount in this function too. 

Also add a test case for this.

Reply:
done